File: Pythonleetcode/Leetcode1551.py
class Solution(object):
    def minOperations(self, n):
        half_length=n//2

        if n%2==1:
            ans = 0
            for i in range(1,half_length + 1):
                ans += i*2
            return ans

        else:
            ans = 0
            for i in range(1,half_length + 1):
                ans += i*2-1
            return ans


print(Solution().minOperations(3))
print(Solution().minOperations(6))

File: CppPractice/Project1/Project1/Leetcode1551.cpp

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int minOperations(int n) {
        int half = n / 2;
        int ans=0;
        if (n%2==0)
        {
            for (int i = 0; i < half; i++)
            {
                ans += (1 + 2 * i);

            }
        }
        else
        {
            for (int i = 0; i < half; i++)
            {
                ans += (2 + 2 * i);

            }
        }
        return ans;

    }
};